echo.py

```diff
@@ -2,21 +2,53 @@
 # -*- coding: utf-8 -*-
 """An enhanced version of the 'echo' cmd line utility"""
 
-__author__ = "???"
+__author__ = "knmarvel"
 
 
-import sys
+import argparse
 
 
 def create_parser():
-    """Creates and returns an argparse cmd line option parser"""
-    pass
+    parser = argparse.ArgumentParser(
+        description="Perform transformation on input text.")
+    parser.add_argument("text",
+                        type=str,
+                        help="text to be manipulated")
+    parser.add_argument('-u',
+                        '--upper',
+                        action='store_const',
```

Comment:
try action = "store_true" to turn this into a boolean with a default of true.  (conveniently gets rid of those uppercase help words at the same time...)